### PR TITLE
ADD: SNLI example

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: finetune/encoding.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ def _encode(self, texts, labels=None, verbose=True):
                 ])
                 token_start = raw_text.find(token.text, token_start)
                 assert len("".join(bpe_toks).replace("</w>", "")) == len(token.text.strip())
-                subtoken_positions = np.cumsum([len(tok.replace("</w>", '')) for tok in bpe_toks]) + token_start
+                subtoken_positions = np.cumsum([len(tok.replace("</w>", ' ')) for tok in bpe_toks]) + token_start
```
Comment (madisonmay) [inline finetune/encoding.py]: sneaky off by one error

File: finetune/utils.py [hunk 2]
```diff
@@ -225,18 +225,21 @@ def finetune_to_indico_sequence(data, labels, none_value=config.PAD_TOKEN):
...
+        
         for sub_str, label in zip(doc, label_seq):
             doc_text += sub_str
+            text = sub_str.strip()
             if label != none_value:
                 doc_annotations.append(
                     {
                         "start": char_loc,
-                        "end": char_loc + len(sub_str),
+                        "end": char_loc + len(text),
```
Comment (madisonmay) [inline finetune/utils.py]: other half of the sneaky off by one error

File: finetune/datasets/multinli.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+class MultiNLI(Dataset):
+
+    def __init__(self, filename=None, **kwargs):
+        super().__init__(filename=(filename or DATA_PATH), **kwargs)
+
+    def download(self):
+        """
+        Download Stanford Sentiment Treebank to enso `data` directory
```
Comment (benleetownsend) [inline finetune/datasets/multinli.py]: I dont think this is the Enso data dir anymore

File: finetune/datasets/multinli.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        """
+        path = Path(self.filename)
+        if path.exists():
```
Comment (benleetownsend) [inline finetune/datasets/multinli.py]: Personally, for a demo like this. I think its nice to do something like
 ```
if path.exists() and hashlib.md5(path).hexdigest() == CHECKSUM:
...
```

Just to mean that if it stops mid-download people don't get terrible results from half a csv, or worse have to dig around finding half a file to manually remove. Just an idea - not critical at all.

File: finetune/encoding.py (hunk 1 above)
Comment (benleetownsend) [reply finetune/encoding.py]: Worth updating the assertion to reflect this. eg.
```
assert len("".join(bpe_toks).replace("</w>", " ")) == len(token.text) 
```

File: setup.py [hunk 5]
```diff
@@ -9,7 +9,6 @@
 REQUIREMENTS = [
     "pandas>=0.23.1",
     "IndicoIo>=1.1.5",
-    "enso>=0.1.6",
```
Comment (benleetownsend) [inline setup.py]: Is it necessary to have the requirements duplicated? Can we not reference the same requirements.txt in the setup.py?

File: tests/test_classifier.py [hunk 6]
```diff
@@ -26,12 +26,12 @@ class TestClassifier(unittest.TestCase):
...
+        'Data', 'Classify', 'SST-binary.csv'
     )
     @classmethod
     def _download_sst(cls):
         """
-        Download Stanford Sentiment Treebank to enso `data` directory
+        Download Stanford Sentiment Treebank to data directory
         """
         path = Path(cls.dataset_path)
         if path.exists():
```
Comment (benleetownsend) [inline tests/test_classifier.py]: Same as above, but 100% just a personal taste thing.

**Pull request merged**